### PR TITLE
refactor: clean up recordings search API endpoints

### PR DESCRIPTION
**Pull request by eddieantonio**
Just various attempts at try to remove the duplication in the two various recording search API endpoints. A lot of it is gone, but there is some duplication that I think is okay to have (or, more precisely, the solution I'm thinking of to get rid of the duplication is just way too much >.<).

**EDIT**: this is what I mean by "some duplication is okay to have": https://github.com/UAlbertaALTLab/recording-validation-interface/compare/refactor/clean-up-recordings-search...DO-NOT-MERGE/template-method-refactor-i-tried?expand=1

### CODE REVIEW

File: validation/views.py [hunk 1]
```diff
@@ -699,3 +644,60 @@ def save_issue(data, user):
...
+    Returns an absolute URL for the compressed audio recording.
+    This can be directly used in an <audio> tag to hear the recording on a webpage!
+    """
+    uri = rec.compressed_audio.url
+    if uri.startswith("/"):
+        # It's a relative URI: build an absolute URI:
+        return request.build_absolute_uri(uri)
+
+    # It's an absolute URI already:
+    assert uri.startswith(("http://", "https://"))
```
Comment (andrewdotn) [inline validation/views.py:685]: Cool, didn’t know you could do this.

File: validation/views.py [hunk 2]
```diff
@@ -699,3 +644,60 @@ def save_issue(data, user):
     )
 
     new_issue.save()
+
+
+def create_recording_result_json(request: HttpRequest, rec: Recording):
```
Comment (andrewdotn) [inline validation/views.py:649]: Maybe make this an instance method, `Recording.serialize()` ?

File: validation/views.py [hunk 3]
```diff
@@ -699,3 +644,60 @@ def save_issue(data, user):
...
+    if uri.startswith("/"):
+        # It's a relative URI: build an absolute URI:
+        return request.build_absolute_uri(uri)
+
+    # It's an absolute URI already:
+    assert uri.startswith(("http://", "https://"))
+    return uri
+
+
+def exclude_known_bad_recordings(recordings: QuerySet):
```
Comment (andrewdotn) [inline validation/views.py:689]: This *could* be a [Manager](https://docs.djangoproject.com/en/3.2/topics/db/managers/), e.g., `Recording.not_known_bad_objects` in addition to `Recording.objects`

**Review by andrewdotn — APPROVED** [review]

**Review by nienna73 — APPROVED** [review]
Thank you!

**Pull request merged**